src/coreclr/System.Private.CoreLib/src/System/Threading/Thread.CoreCLR.cs

```diff
@@ -189,22 +194,22 @@ public static void SpinWait(int iterations)
         /// </summary>
         public bool IsBackground
         {
-            get => IsBackgroundNative();
+            get => GetIsBackground(GetNativeHandle()) != Interop.BOOL.FALSE;
```

Comment:
Can the following happen now?

1. We get the native thread handle here.
2. This happened to be the last reference of this managed thread object. The thread object is no longer reported to the GC once we have the native thread handle.
3. GC runs right before the call of the native QCall method
4. Both managed and native thread objects are cleaned up.
5. Execution of this method resumes. The native QCall method gets a dangling pointer. Intermittent crash.

Reply:
Yes, that was a concern I had. I looked at some uses of `GetNativeHandle()` and convinced myself that the other places weren't handling this so decided to accept it as "valid". However, looking at it again each use case is "protected" by a local `lock(this)` or one in a high-level caller function. I assume this is effectively performing a "KeepAlive()".

I'm inclined to put a `GC.KeepAlive()` in all the relevant uses, fair?

Reply:
`GC.KeepAlive` sounds good.